src/Microsoft.AspNetCore.Hosting/Internal/HostingApplication.cs

```diff
@@ -98,6 +152,13 @@ public void DisposeContext(Context context, Exception exception)
             // If startTimestamp is 0, don't call GetTimestamp, likely don't need the value
             var currentTimestamp = (startTimestamp != 0) ? Stopwatch.GetTimestamp() : 0;
 
+#if NETSTANDARD2_0
+            // TODO: Ability to set duration directly or set end time from timestamp
+            context.Activity?
+                .SetEndTime(DateTime.UtcNow)
+                .Stop();
+#endif
+
             // To keep the hot path short we defer logging to non-inlines
             if (exception == null)
```

Comment:
I'm not sure what happens now if request is cancelled, will exception be TaskCancelledException?
End (or Stop, depending on what we decide in other discussion) event must be sent for all requests including faulted or cancelled.

Imagine in your service you've got only Begin event and never got End. Does it mean there was an exception or user cancelled the request? Or log record was lost somehow because of bug or infrastructure problem. Or request is still being processed? Always sending End/Stop event helps consumer to do logging without guessing what happened.

End/Stop event payload must also carry TaskStatus that helps consumer to understand what happened with the request. It does not break any existing consumers.

I see also that `IsEnabled(RecordEndRequestDiagnostics)` is checked before sending the event. As I mentioned before, it's hard to imagine that someone listens to Begin, but not listens to End, so I think this check may be removed.